stonesoup/plotter.py

```diff
@@ -202,14 +202,17 @@ def plot_tracks(self, tracks, mapping, uncertainty=False, particle=False, **kwar
 
         # Plot tracks
         for track in tracks:
-            self.ax.plot([state.state_vector[mapping[0]] for state in track],
-                         [state.state_vector[mapping[1]] for state in track],
-                         **tracks_kwargs)
+            line = self.ax.plot([state.state_vector[mapping[0]] for state in track],
+                                [state.state_vector[mapping[1]] for state in track],
+                                **tracks_kwargs)
+
+        # Assuming a single track or all plotted as the same colour then the following will work.
+        # Otherwise will just render the final track colour.
+        tracks_kwargs['color'] = plt.getp(line[0], 'color')
```

Comment:
Not sure if this is an issue worth fixing but this means uncertainty ellipses are also plotted as the same colour as the final track. Could be fixed by making all ellipses a default colour?

Reply:
It's a good thing that the ellipses match the track colour, no?

Reply:
Yes, sorry I meant when there are multiple tracks with different colours all the ellipses will have the colour of the final track, not the track they are attached to.

Reply:
Oh, I see. That might look a bit odd. 

I imagine there must be a way of finding out, ahead of time or separately, the order in which colours appear, and then passing that to the legend/ellipse plotter.

Reply:
You could use the same `plt.getp(line[0], 'color')` in the loop above, creating a dictionary of colours keyed by the track, and then use that for the ellipse plotting below.

https://github.com/dstl/Stone-Soup/pull/394/files#diff-40bed2921b0b0211da770554c467db7cc29b9e0f0fa0c436191a7f1b50f95458R232

Reply:
That link's not taking me anywhere...

Reply:
Yeah, not working for me now either. Was just linking to the line below where the color is passed for the ellipse.

https://github.com/dstl/Stone-Soup/blob/0fc6be4054da5b63c388517b51cb224cfa5186a5/stonesoup/plotter.py#L232